### PR TITLE
lightning-pool: init at version 0.4.1

### PR DESCRIPTION
**Pull request by ProofOfKeags**
###### Motivation for this change
This adds Lightning Pool Client to nixpkgs

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -27541,6 +27541,8 @@ in
 
   ledger-live-desktop = callPackage ../applications/blockchains/ledger-live-desktop { };
 
+  lightning-pool = callPackage ../applications/blockchains/lightning-pool.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:27544]:
```suggestion
  lightning-pool = callPackage ../applications/blockchains/lightning-pool { };
```

File: pkgs/applications/blockchains/lightning-pool.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchFromGitHub {
+    owner = "lightninglabs";
+    repo = "pool";
+    rev = "v${version}";
+    sha256 = "0wn9jp61hi2wdlxfk1wsyngxs4fsgjpbgjzsy1z2hj9l3ppig1kf";
+  };
+
+  vendorSha256 = "06ama1z3ksq6125s0j6ddwc3ljbqfsyhmdx5y3knchyfjx3m9m15";
+
+  subPackages = ["cmd/pool" "cmd/poold"];
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/lightning-pool.nix:19]:
```suggestion
  subPackages = [ "cmd/pool" "cmd/poold" ];
```

**Review by SuperSandro2000 — COMMENTED** [review]
Please put the file into a directory.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (ProofOfKeags) [reply pkgs/top-level/all-packages.nix:27544]: Same here as with loop. This is not the way that lnd was done. I am happy to do as you ask but I was going based off of how I saw other related packages in the file tree.

Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:27544]: All other packages who do it like this in this file are packaged wrong.

Comment (ProofOfKeags) [reply pkgs/top-level/all-packages.nix]: fixed by 8c9061b

**Comment by SuperSandro2000** [comment]
Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME``.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please squash all the package related commits.

**Pull request merged**